File: Php/tests/AppTests/Cashgame/Running/StatusTableModelTests.php
namespace tests\AppTests\Cashgame\Running{

	use entities\Cashgame;
	use entities\CashgameResult;
	use entities\Player;
	use tests\Fakes\TimerFake;
	use DateTime;
	use entities\Homegame;
	use tests\UnitTestCase;
	use app\Cashgame\Running\StatusTableModel;
	use tests\TestHelper;

	class StatusTableModelTests extends UnitTestCase {

		/** @var Homegame */
		private $homegame;
		/** @var Cashgame */
		private $cashgame;

		function setUp(){
			homegame = new Homegame();
			cashgame = new Cashgame();
		}

		function test_StatusModels_CashgameWithOnePlayer_FirstItemIsCorrectType(){
			cashgame.setResults(array(new CashgameResult()));

			$sut = getSut();

			assertEqual(1, count($sut.statusModels));
			assertIsA($sut.statusModels[0], 'app\Cashgame\Running\StatusItemModel');
		}

		function test_StatusModels_CashgameWithTwoPlayers_HasTwoItems(){
			cashgame.setResults(array(new CashgameResult(), new CashgameResult()));

			$sut = getSut();

			assertEqual(2, count($sut.statusModels));
		}

		function test_TotalBuyin_CashgameWithTwoPlayers_IsSumOfBuyins(){
			cashgame.setTurnOver(1);

			$sut = getSut();

			assertEqual("$1", $sut.totalBuyin);
		}

		function test_TotalStacks_CashgameWithTwoPlayers_IsSumOfCurrentStacks(){
			cashgame.setTotalStacks(1);

			$sut = getSut();

			assertEqual("$1", $sut.totalStacks);
		}

		function test_StatusModels_CashgameWithTwoPlayers_IsSortedByWinningsDescending(){
			cashgame.setStartTime(new DateTime());
			$player1 = new Player();
			$player1.setDisplayName('a');
			$result1 = new CashgameResult();
			$result1.setPlayer($player1);
			$result1.setWinnings(1);
			$player2 = new Player();
			$player2.setDisplayName('b');
			$result2 = new CashgameResult();
			$result2.setPlayer($player2);
			$result2.setWinnings(2);
			cashgame.setResults(array($result1, $result2));

			$sut = getSut();

			$results = $sut.statusModels;
			assertIdentical('b', $results[0].name);
			assertIdentical('a', $results[1].name);
		}

		function getSut(){
			return new StatusTableModel(homegame, cashgame, false, new TimerFake());
		}

	}

}

File: Web.Tests/ModelTests/CashgameModels/Running/StatusTableModelTests.cs
using System;
using System.Collections.Generic;
using Core.Classes;
using NUnit.Framework;
using Tests.Common;
using Web.Models.CashgameModels.Running;

namespace tests{

	public class StatusTableModelTests : MockContainer{

		private Homegame _homegame;
		private Cashgame _cashgame;

        [SetUp]
		public void SetUp(){
			_homegame = new Homegame();
			_cashgame = new Cashgame();
		}

		[Test]
        public void StatusModels_CashgameWithOnePlayer_FirstItemIsCorrectType(){
			_cashgame.Results = new List<CashgameResult>{new CashgameResult()};

			var sut = GetSut();

			Assert.AreEqual(1, sut.StatusModels.Count);
		    Assert.IsInstanceOf<StatusItemModel>(sut.StatusModels[0]);
        }

		[Test]
        public void StatusModels_CashgameWithTwoPlayers_HasTwoItems(){
			_cashgame.Results = new List<CashgameResult>{new CashgameResult(), new CashgameResult()};

			var sut = GetSut();

			Assert.AreEqual(2, sut.StatusModels.Count);
		}

		[Test]
        public void TotalBuyin_CashgameWithTwoPlayers_IsSumOfBuyins(){
			_cashgame.Turnover = 1;

			var sut = GetSut();

			Assert.AreEqual("$1", sut.TotalBuyin);
		}

		[Test]
        public void TotalStacks_CashgameWithTwoPlayers_IsSumOfCurrentStacks(){
			_cashgame.TotalStacks = 1;

			var sut = GetSut();

			Assert.AreEqual("$1", sut.TotalStacks);
		}

		[Test]
        public void StatusModels_CashgameWithTwoPlayers_IsSortedByWinningsDescending(){
			_cashgame.StartTime = new DateTime();
			var player1 = new Player {DisplayName = "a"};
		    var result1 = new CashgameResult {Player = player1, Winnings = 1};
		    var player2 = new Player {DisplayName = "b"};
		    var result2 = new CashgameResult {Player = player2, Winnings = 2};
		    _cashgame.Results = new List<CashgameResult>{result1, result2};

			var sut = GetSut();

			var results = sut.StatusModels;
			Assert.AreEqual("b", results[0].Name);
			Assert.AreEqual("a", results[1].Name);
		}

		private StatusTableModel GetSut(){
			return new StatusTableModel(_homegame, _cashgame, false, TimeProviderMock.Object);
		}

	}

}